community/modules/scheduler/schedmd-slurm-gcp-v6-controller/modules/slurm_files/scripts/slurmsync.py

```diff
@@ -535,6 +535,55 @@ def sync_maintenance_reservation(lkp: util.Lookup) -> None:
       create_reservation(lkp, res_name, node, start_time)
 
 
+def delete_maintenance_job(job_name: str) -> None:
+    util.run(f"scancel --name={job_name}")
+
+
+def create_maintenance_job(job_name: str, node: str) -> None:
+    util.run(f"sbatch --job-name={job_name} --nodelist={node} /slurm/custom_scripts/perform_maintenance.sh")
+
+
+def get_slurm_maintenance_job(lkp: util.Lookup) -> Dict[str, str]:
+    maintenance_job = {}
+
+    for job in lkp.get_jobs():
+        if job.name is None or job.required_nodes is None or job.job_state is None:
+          continue
+
+        if job.name != f"{job.required_nodes}_maintenance":
+          continue
+
+        if job.job_state != "PENDING":
+          continue
+
+        maintenance_job[job.name] = job.required_nodes
+
+    return maintenance_job
+
+
+def sync_opportunistic_maintenance(lkp: util.Lookup) -> None:
+    upc_maint_map = get_upcoming_maintenance(lkp)  # map job_name -> (node_name, time)
+    log.debug(f"upcoming-maintenance-vms: {upc_maint_map}")
+
+    curr_maintenance_job_map = get_slurm_maintenance_job(lkp)  # map job_name -> node.
+    log.debug(f"curr-maintenance-job-map: {curr_maintenance_job_map}")
+
+    del_maintenance_job_set = set(curr_maintenance_job_map.keys() - upc_maint_map.keys())
```

Comment:
This code will go through significant refactoring after you will need to account for enabled/disabled opportunistic maintenance per nodeset. 
It's fine to land it as is , since it's effectively a dead code. 
But I want to bring your attention to the fact that current review is kinda pointless.

Reply:
I updated the code to include enable/disable opportunistic maintenance per nodeset level. 
lmkwyt.